src/cmd/run.js

```diff
@@ -322,6 +322,8 @@ export class ExtensionRunner {
 
   run(profile: FirefoxProfile): Promise<FirefoxProcess> {
     const {firefoxApp, firefox} = this;
-    return firefoxApp.run(profile, {firefoxBinary: firefox});
+    return firefoxApp.run(
+        profile, {firefoxBinary: firefox, binaryArgs: ['-jsconsole']}
```

Comment:
@aniketkudale Sure! 

the new "web-ext run" cli option is going to flow through the web-ext sources like:
```
"the new web-ext run cli option" (src/program.js) ->
  "export default async function run" (src/cmd/run.js) ->
    "ExtensionRunner constructor" (src/cmd/run.js) ->
      "ExtensionRunner run method" (src/cmd/run.js)
```

follow some additional details of the changes needed at each of the steps sketched above:

- "the new web-ext run cli option" (src/program.js)
  - the `yargs` type of the option should be a `"boolean"` instead of `"string"` (and, at least in my opinion, it would be probably better to rename it into `--browser-console`, which is how the tool is called in the devtools menu)
- "export default async function run" (src/cmd/run.js) ->
  - the inew option should be added to the `type CmdRunParams` (change needed to pass the flowtype checks), with type `boolean`
  - the new option should be added to the "export default async function run" (change needed to be able to access the new option from the function body), and its default value should be `false`
  - pass the option into the `ExtensionRunner` constructor params
- "ExtensionRunner constructor options" (src/cmd/run.js)
  - the new option should be added to the `type ExtensionRunnerParams` (line 263)
   and to the fields of the `ExtensionRunner` class (line 279), with type `boolean` 
  - the new option should be added to the `ExtensionRunner` constructor params and stored into `this.myVarName` (so that you can use it in the `ExtensionRunner` run method)
 - "`ExtensionRunner` `run` method" (src/cmd/run.js)
   - now you check `this.myVarName` in the `ExtensionRunner` run method (line 331) and
   add `--jsconsole` to the `binaryArgs` if needed (when `this.myVarName` is defined and its value is `true`)

This new cli option will require mostly the same changes needed for #709 (which is still in progress, and I'm going to review asap)

Let me know if the above details are helpful or if something is still missing from the above list.